packages/ui/src/PhoneNumberField.tsx

```diff
@@ -21,11 +21,16 @@ export const PhoneNumberField: FC<PhoneNumberFieldProps> = ({
   useEffect(() => {
     setLocalValue(value || "");
     setError(errorText);
-  }, [value, errorText]);
+    // If the value is undefined on mount, initialize to an empty string to ensure first input
+    // is recorded
+    if (value === undefined) {
```

Comment:
I think you can get away without this call to `onChange` given the improvements to error handling below. I checked out your branch and tried commenting out lines 26-28 and we still got the behavior we want. This would then prevent setting phone numbers to empty strings if we want them to stay undefined for some reason.

Reply:
ah good catch, you're right!